### PR TITLE
Add support for symfony 5 and dbal 2.10

### PR DESCRIPTION
**Pull request by paillechat**

### CODE REVIEW

File: src/DependencyInjection/Configuration.php [hunk 1]
```diff
@@ -11,9 +11,15 @@ final class Configuration implements ConfigurationInterface
     /** {@inheritdoc} */
     public function getConfigTreeBuilder(): TreeBuilder
     {
-        $builder = new TreeBuilder();
-
-        $root = $builder->root('lamoda_enum');
+        $version = \PackageVersions\Versions::getVersion('symfony/dependency-injection');
```
Comment (scaytrase) [inline src/DependencyInjection/Configuration.php]: This thing is broken when it comes to transitive dependencies (https://github.com/Ocramius/PackageVersions/issues/116) . The common approach is to rely on `method_exists` to determine current API version

https://github.com/doctrine/DoctrineBundle/commit/e711bed5a82437262aeade06373fcd5c81a2fd94

**Review by scaytrase — CHANGES_REQUESTED** [review]
You should also update the testing matrix in order to add `5.0.*` to ensure that everything works correctly with newer symfony

**Comment by scaytrase** [comment]
The builds are broken

File: src/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -11,9 +11,14 @@ final class Configuration implements ConfigurationInterface
     /** {@inheritdoc} */
     public function getConfigTreeBuilder(): TreeBuilder
     {
-        $builder = new TreeBuilder();
-
-        $root = $builder->root('lamoda_enum');
+        $name = 'lamoda_enum';
+        if (method_exists(TreeBuilder::class, 'getRootNode')) {
```
Comment (omnilight) [inline src/DependencyInjection/Configuration.php]: Это условие всегда будет true, метод же есть в обеих версиях, отличие в сигнатуре

Comment (paillechat) [reply src/DependencyInjection/Configuration.php]: Да, поменял проверку.

File: composer.json [hunk 3]
```diff
@@ -13,14 +13,14 @@
...
     "php": "^7.1",
     "paillechat/php-enum": "^1.2 || ^2.1",
-    "symfony/http-kernel": "^3.4 || ^4.0",
-    "symfony/dependency-injection": "^3.4 || ^4.0",
-    "symfony/config": "^3.4 || ^4.0",
-    "doctrine/dbal": "^2.4"
+    "symfony/http-kernel": "^3.4 || ^4.0 || ^5.0",
+    "symfony/dependency-injection": "^3.4 || ^4.0 || ^5.0",
+    "symfony/config": "^3.4 || ^4.0 || ^5.0",
+    "doctrine/dbal": ">=2.4"
```
Comment (scaytrase) [inline composer.json]: this change allows not only dbal 2.10, but also any of 3.x if released in future. was it intentional?

**Review by scaytrase — APPROVED** [review]
LGTM 👍 One minor quesiton left

File: composer.json (hunk 3 above)
Comment (paillechat) [reply composer.json]: No, it was not my intention as I can't assume which changes will come in dbal3. 
What is your suggestion? 
Strongly define 2.4 || 2.10, or reduce a range using interval like >=2.4 <3?

Comment (scaytrase) [reply composer.json]: `^2.4` was already permitting 2.10 to come

https://jubianchi.github.io/%5E2.4/2.10.0

**Comment by scaytrase** [comment]
Thank you @paillechat

**Pull request merged**